generators/chipyard/src/main/scala/TestSuites.scala

```diff
@@ -109,37 +113,41 @@ class TestSuiteHelper
   */
   def addRocketTestSuites(implicit p: Parameters) = {
     val xlen = p(XLen)
-    p(RocketTilesKey).find(_.hartId == 0).map { tileParams =>
-      val coreParams = tileParams.core
-      val vm = coreParams.useVM
-      val env = if (vm) List("p","v") else List("p")
-      coreParams.fpu foreach { case cfg =>
-        if (xlen == 32) {
-          addSuites(env.map(rv32uf))
-          if (cfg.fLen >= 64)
-            addSuites(env.map(rv32ud))
-        } else {
-          addSuite(rv32udBenchmarks)
-          addSuites(env.map(rv64uf))
-          if (cfg.fLen >= 64)
-            addSuites(env.map(rv64ud))
+    p(TilesLocated(InSubsystem)).find(_.tileParams.hartId == 0).map {
+      case tp: RocketTileAttachParams => {
```

Comment:
These seem quite redundant to me but that refactor is definitely for another PR.

Reply:
I agree with both statements